### ISSUE TITLE
Bug with compareAttribute

### ISSUE THREAD

**Issue by Inkognitoo** [yiisoft/yii2#16645]
### What steps will reproduce the problem?

```
<?php


namespace app\models;


use yii\base\Model;

class Test extends Model
{
    public $a;
    public $b;

    public function rules()
    {
        return [
            [['a', 'b'], 'safe'],
            [['a'], 'compare', 'compareAttribute' => 'b'],
        ];
    }
}
```
```
<?php

use app\models\Test;
use yii\widgets\ActiveForm;

$model = new Test();

$form = ActiveForm::begin();
echo $form->field($model, 'a')->textInput();
echo $form->field($model, 'b')->textInput();
ActiveForm::end();
```

### What is the expected result?

Error if fields not compare and no error if they are compare

### What do you get instead?

Error anyway

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15
| PHP version    | 7.2.7 
| Operating system |

If you add some extra rules, like:
```
public function rules()
    {
        return [
            [['a', 'b'], 'safe'],
            [['a', 'b'], 'string'],
            [['a'], 'compare', 'compareAttribute' => 'b'],
        ];
    }
```

It's work fine

**Comment by mgrechanik** [comment]
#15826

**Comment by Inkognitoo** [comment]
@mgrechanik okay, thank you :ok_hand:

**Comment by mgrechanik** [comment]
check your issue with the current developer version

**Comment by Inkognitoo** [comment]
@mgrechanik I see your pull request and understand about bug fix in 2.0.16
